transport/server.cc

```diff
@@ -897,13 +898,36 @@ future<std::unique_ptr<cql_server::response>> cql_server::connection::process_st
     std::unique_ptr<cql_server::response> res;
     if (auto& a = client_state.get_auth_service()->underlying_authenticator(); a.require_authentication()) {
         _authenticating = true;
-        res = make_autheticate(stream, a.qualified_java_name(), trace_state);
+        auto opt_user = co_await a.authenticate([this]() -> future<std::optional<auth::certificate_info>> {
+            auto dn_info = co_await tls::get_dn_information(this->_fd);
+            if (dn_info) {
+                co_return auth::certificate_info{ dn_info->subject, [this]() -> future<std::string> {
+                    std::stringstream ss;
+                    auto altnames = co_await tls::get_alt_name_information(this->_fd);
+                    auto delim = "";
+                    for (auto& a : altnames) {
+                        ss << delim << a;
+                        delim = ",";
+                    }
+                    co_return ss.str();
```

Comment:
This can be written as: `co_return fmt::format("{}", fmt::join(altnames, ","));`. We're trying to move away from iostreams.

Reply:
Changed.